modules/somalier/extract/main.nf

```diff
@@ -0,0 +1,40 @@
+
+process SOMALIER_EXTRACT {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::somalier=0.2.15" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/somalier:0.2.15--h37c5b7d_0':
+        'quay.io/biocontainers/somalier:0.2.15--h37c5b7d_0' }"
+
+
+    input:
+    tuple val(meta), path(bam), path(bai)
+    path(ref)
+    path(refidx)
+    path(sites)
+
+    output:
+    tuple val(meta), path("*.somalier.extract/*.somalier"),             emit: extract
+    path "versions.yml",                                                emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    """
+    somalier extract ${args} -d ${prefix}.somalier.extract \
```

Comment:
```suggestion
    somalier extract ${args} \\
    -s ${prefix} \\
```

Reply:
I don't think `-d` is necessary given it has a default `-d .`

```
  -d, --out-dir=OUT_DIR      path to output directory (default: .)
```

This way you can just pick up all the files with a simple `*.somalier`

Reply:
I am _presuming_ `-s` corresponds to the sample name prefix that will correspond to the output files, but I may be wrong here (I'm not sure what `digest` means in the help message...

Reply:
Not really, the -s is the same as the --sites, which is the "sites" input file. But I totally agree with the "-d" part. I've now changed the code accordingly. This came from an older code where I had to provide an additional folder, since the somalier was/is naming the files based on the sample names from the bam/vcf files and not the filenames themselves, which I needed at that point... In any case, it is changed now :)